types/react-router-redux/index.d.ts

```diff
@@ -1,63 +1,65 @@
-// Type definitions for react-router-redux 4.0
-// Project: https://github.com/rackt/react-router-redux
-// Definitions by: Isman Usoh <http://github.com/isman-usoh>
-//                 Noah Shipley <https://github.com/noah79>
-//                 Dimitri Rosenberg <https://github.com/rosendi>
-//                 Karol Janyst <https://github.com/LKay>
-//                 Dovydas Navickas <https://github.com/DovydasNavickas>
+// Type definitions for react-router-redux 5.0
+// Project: https://github.com/ReactTraining/react-router/tree/master/packages/react-router-redux
+// Definitions by: Huy Nguyen <https://github.com/huy-nguyen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// TypeScript Version: 2.2
 
-import { Action, Middleware, Store } from "redux";
-import { History, Location, LocationDescriptor } from "history";
+import {
+    Store,
+    Dispatch,
+    Action,
+    Middleware
+} from 'redux';
+import {
+    History,
+    Location,
+    Path,
+    LocationState,
+    LocationDescriptorObject
+} from 'history';
+import * as React from 'react';
 
-export const CALL_HISTORY_METHOD: string;
-export const LOCATION_CHANGE: string;
-
-export interface LocationActionPayload {
-    method: string;
-    args?: any[];
+export interface ConnectedRouterProps<State> {
+    store?: Store<State>,
+    history?: History
 }
+export class ConnectedRouter<State> extends React.Component<ConnectedRouterProps<State>, {}> {}
 
-export interface RouterAction extends Action {
-    payload?: LocationActionPayload;
-}
+export const LOCATION_CHANGE = '@@router/LOCATION_CHANGE'
```

Comment:
@LKay is there a reason why `declare const LOCATION_CHANGE: string` is better? Is it just for simplicity? Defining it as `export const LOCATION_CHANGE = '@@router/LOCATION_CHANGE'` makes `LOCATION_CHANGE` have the literal type of `'@@router/LOCATION_CHANGE'` instead of just a string.